Agents.md

```
@@ -0,0 +1,1352 @@
+
+// ⚠️ Important: Variables won't work with direct indexing
+let eq_char : Char = '='
+// s.code_unit_at(0) == eq_char // ❌ Won't compile - eq_char is not a literal, lhs is UInt while rhs is Char
+// Use: s.code_unit_at(0) == '=' or s.get_char(0) == Some(eq_char)
+let bytes = @encoding/utf8.encode("中文") // utf8 encode package is in stdlib
+assert_true(bytes is [0xe4, 0xb8, 0xad, 0xe6, 0x96, 0x87])
+let s2 : String = @encoding/utf8.decode(bytes) // decode utf8 bytes back to String
+assert_true(s2 is "中文")
+```
+
+#### String Interpolation
```

markdownlint-cli2 (0.18.1)

Heading levels should only increment by one level at a time
Expected: h3; Actual: h4

(MD001, heading-increment)